Assets/Shopify/UIToolkit/Theme Controllers/ThemeControllerBase.cs

```diff
@@ -13,25 +14,47 @@ public abstract class ThemeControllerBase : MonoBehaviour, IShopCredentials {
         /// The Shop Domain to connect to, in the form of "myshop.myshopify.com"
         /// </summary>
         public string ShopDomain {
-            get { return _shopDomain; }
+            get { 
+                if (_shopDomain == null) {
+                    _shopDomain = EditorPrefs.GetString(ShopDomainEditorPrefsKey);
+                }
+                return _shopDomain; 
+            }
             set {
                 _shopDomain = value;
                 InvalidateClient();
+                EditorPrefs.SetString(ShopDomainEditorPrefsKey, value);
             }
         }
         [SerializeField] private string _shopDomain;
+        private string ShopDomainEditorPrefsKey {
+            get {
+                return GetInstanceID() + ":shopDomain";
+            }
+        }
 
         /// <summary>
         /// The Storefront Access Token used to authenticate with the connected shop
         /// </summary>
         public string AccessToken {
-            get { return _accessToken; }
+            get { 
+                if (_accessToken == null) {
+                    _accessToken = EditorPrefs.GetString(AccessTokenEditorPrefsKey);
+                }
+                return _accessToken; 
+            }
             set {
                 _accessToken = value;
                 InvalidateClient();
+                EditorPrefs.SetString(AccessTokenEditorPrefsKey, value);
             }
         }
         [SerializeField] private string _accessToken;
+        private string AccessTokenEditorPrefsKey {
+            get {
+                return GetInstanceID() + ":accessToken";
```

Comment:
As per my comment from before, the goal is to save credentials across instances rather than for an individual instance (you are right that the serialization takes care of it per instance). So the key should be something like `shopify_buy_last_valid_access_token` or something (note the namespacing to avoid conflicts), especially as EditorPrefs are shared by all of the different scripts you might have installed.